src/server/string_family.cc

```diff
@@ -666,6 +688,74 @@ void StringFamily::GetSet(CmdArgList args, ConnectionContext* cntx) {
   return (*cntx)->SendNull();
 }
 
+void StringFamily::GetEx(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 1);
+
+  DbSlice::ExpireParams exp_params;
+  int64_t int_arg = 0;
+
+  for (size_t i = 2; i < args.size(); i++) {
+    string_view cur_arg = ArgS(args, i);
+    bool is_ms = (cur_arg == "PX" || cur_arg == "PXAT");
+    i++;
+    if (i >= args.size()) {
+      return (*cntx)->SendError(kSyntaxErr);
+    }
+
+    string_view ex = ArgS(args, i);
+    if (!absl::SimpleAtoi(ex, &int_arg)) {
+      return (*cntx)->SendError(kInvalidIntErr);
+    }
+
+    if (int_arg <= 0) {
+      return (*cntx)->SendError(InvalidExpireTime("getex"));
+    }
+
+    if (cur_arg == "EXAT" || cur_arg == "PXAT") {
+      exp_params.absolute = true;
+      int_arg = CalculateAbsTime(int_arg, is_ms);
+      if (int_arg < 0) {
+        return (*cntx)->SendStored();
+      }
+    }
+
+    if (!is_ms && int_arg >= kMaxExpireDeadlineSec) {
```

Comment:
please remove this check. it's wrong the way it lays here and `UpdateExpire` already checks the arguments for correctness.